dll/win32/shell32/CDefView.cpp

```diff
@@ -3574,6 +3574,11 @@ HRESULT WINAPI CDefView::Drop(IDataObject* pDataObject, DWORD grfKeyState, POINT
     ImageList_DragLeave(m_hWnd);
     ImageList_EndDrag();
 
+    if (GetAsyncKeyState(VK_SHIFT) < 0)
+        grfKeyState |= MK_SHIFT;
+    else
+        grfKeyState &= ~MK_SHIFT;
+
```

Comment:
It's called by ole32!DoDragDrop, which seems to send this using `OLEDD_GetButtonState()`.
Most likely the same root-cause as the other change, where you needed to change `GetKeyState` to `GetAsyncKeyState`.
(in a message loop, keystate is not updated properly?)

Reply:
From testing (at least dropping a file on the desktop) this seems to be updated properly.
Was there a specific case fixed by this?

Reply:
This is correctly updated in the normal 'drag drop' case, but not when testing this:
* Drag 1 file to the trash bin with shift held
* Drag a second file to the trash bin, explorer still thinks shift key is held down.

Reply:
I have removed these controversial changes for now in hopes that this can get committed more easily. The remaining changes fix about 90% of what I was trying to accomplish by mostly fixing the "shift-delete" file problem. I plan to address the remaining issues in another Jira ticket and PR. Thanks for everyone's help with this. Doug Lyons